File: Data Structures & Algorithms/sort-an-array/submission-1.swift
class Solution {
    func sortArray(_ nums: [Int]) -> [Int] {
        if nums.count == 0 {
            return nums
        } else {
            return self.sortArray(nums[1...].filter { $0 <= nums[0] }) + [nums[0]] + self.sortArray(nums[1...].filter { $0 > nums[0] })
        }
    }
}

File: Data Structures & Algorithms/is-palindrome/submission-1.rs
impl Solution {
    pub fn is_palindrome(s: String) -> bool {
        let ss = s.to_lowercase();
        let mut iter1 = ss.chars().filter(|c| c.is_alphanumeric());
        let mut iter2 = ss.chars().filter(|c| c.is_alphanumeric()).rev();
        while let Some(c) = iter1.next() {
            if Some(c) != iter2.next() {
                return false;
            }
        }
        return true;
    }
}